web_src/js/modules/fetch.js

```diff
@@ -0,0 +1,31 @@
+const {csrfToken} = window.config;
+
+function request(url, {headers, json, ...other} = {}) {
+  return window.fetch(url, {
+    headers: {
+      'x-csrf-token': csrfToken,
+      ...(json !== undefined && {'content-type': 'application/json'}),
```

Comment:
It also needs to support FormData

Reply:
Yes, can add a `formData` property for it.

Reply:
Or better, let's just automatically set `content-type` when `body` is a formdata instance, e.g. API like https://github.com/sindresorhus/ky#sending-form-data.

Reply:
How about naming it "data", and check whether it is a FormData class?

Reply:
If we go with a `data`, we may as well handle json through it. That's what I was actually doing in other projects, e.g. typecheck `data`, when it is non-string, encode as json, but this would not allow sendinging plain strings with json encoding.

Reply:
Another type to consider is urlencoded: https://stackoverflow.com/a/53189376/808699, but we can just typecheck it with `URLSearchParams`.

Reply:
> when it is non-string, encode as json, but this would not allow sending plain strings with json encoding, thought I guess that's a use case we likely don't need.

If in case it needs to be supported, we can introduce a class `BodyString` and do :  `data: BodyString('content')`

Reply:
Done in https://github.com/go-gitea/gitea/pull/26985/commits/11eb0d34fc4ab692ab4eaf7769e07efe04b53205, It also supports string json.

Reply:
And from another aspect, I do not see we really need to support encoding "int"/"string". There is no such use case and backend doesn't really support it.

So I guess at the moment, just do not support int/string/bool (throw error), etc. Only support object/FormData, until there is a clear use case.

Reply:
I guess I will just make it accept only object and array for json encoding.

Reply:
Done in https://github.com/go-gitea/gitea/pull/26985/commits/c02f5bd70069a9ab350c4e515a24bee0049e62a9. One can still pass in `body` for manual usage in case it is ever needed.